dxr/plugins/clang/needles.py

```diff
@@ -35,24 +30,50 @@ def needles(condensed, name, suffix='', kind=None, subkind=None, keys=('name', '
          ``condensed``. Defaults to the value of ``name``.
     :arg subkind: The value of the 'kind' key to insist on in things found
         within ``condensed[kind]``. None for no insistence.
-    :arg keys: The keys that should be in the final needle value
+    :arg include_qualname: Whether we should include the qualified name in the
+        returned needle value dict
+    :arg include_typeless_qualname: Whether we should include the qualname but
+        with arg types stripped off, e.g. turning functions from Foo::bar(int)
+        to Foo::bar, so people don't need to know the types to search for
+        them. ``include_qualname`` must also be true to enable this.
 
     """
     kind = kind or name
     matches_subkind = (lambda entity: entity.get('kind') == subkind if subkind
                        else lambda entity: True)
+
+    def names(entity):
+        """Return the name, qualname, and type-free qualname of an entity.
+
+        This way, Moo::foo(int) can be found as "foo", "Moo::foo" or
+        "Moo:foo(int)".
+
+        """
+        value = {'name': entity['name']}
+        if include_qualname:
+            qualname = value['qualname'] = entity['qualname']
+            if include_typeless_qualname:
+                try:
+                    # This is good enough for functions but probably not for
+                    # templates yet:
+                    offset = qualname.index('(')
+                except ValueError:
+                    pass
+                else:
```

Comment:
Is there a benefit to using the else clause here rather than putting its line into the try section?
At first I thought this was an indentation bug since I was looking at the two if's the else is nested within.